### PR TITLE
fix: pushing and removing errors from debugger for js object

### PR DESCRIPTION
**Pull request by ApekshaBhosale**
## Description
Pushing all parsing errors to the debugger. Changed eval to evaluate to get all errors

Fixes # (issue)

part of 6872 - 
- In some cases, when typing a random character in the JS, invalid JS is not detected and the JSObject does not show an error
- If the JSObject is in an unparseable state, we need to show an error that none of the functions and data will work. Highlight red around the Editor and show an error state for the run button (need design here)

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: pushing-errors-to-debugger 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.76 **(-0.04)** | 36.56 **(0)** | 33.9 **(-0.03)** | 55.31 **(-0.04)**
 :green_circle: | app/client/src/components/editorComponents/JSResponseView.tsx | 43.84 **(0.76)** | 20.78 **(0.49)** | 0 **(0)** | 46.38 **(0.48)**
 :red_circle: | app/client/src/constants/messages.ts | 74.66 **(-0.06)** | 100 **(0)** | 24.43 **(-0.19)** | 81.1 **(-0.35)**
 :green_circle: | app/client/src/pages/Editor/JSEditor/Form.tsx | 48 **(1.33)** | 85.71 **(0)** | 0 **(0)** | 50 **(1.16)**
 :red_circle: | app/client/src/sagas/EvaluationsSaga.ts | 20.71 **(-0.89)** | 1.56 **(-0.03)** | 10 **(0)** | 23.81 **(-1.01)**
 :red_circle: | app/client/src/sagas/JSPaneSagas.ts | 20.39 **(0.13)** | 1.54 **(0)** | 7.14 **(-0.55)** | 24.41 **(0)**
 :red_circle: | app/client/src/selectors/commentsSelectors.ts | 78.23 **(-1.61)** | 54.41 **(-2.94)** | 70 **(0)** | 84.09 **(-2.27)**
 :red_circle: | app/client/src/workers/evaluate.ts | 90.7 **(-4.36)** | 81.82 **(-3.18)** | 80 **(-8.89)** | 90.48 **(-4.46)**
 :red_circle: | app/client/src/workers/evaluationUtils.ts | 74.37 **(-6.83)** | 61.48 **(-6)** | 76.19 **(-1.86)** | 72.53 **(-7.37)**</details>

### CODE REVIEW

File: app/client/src/constants/messages.ts [hunk 1]
```diff
@@ -415,11 +415,18 @@ export const PARSE_JS_FUNCTION_ERROR = (message: string) =>
...
+
 // Import/Export Application features
 export const IMPORT_APPLICATION_MODAL_TITLE = () => "Import Application";
 export const DELETE_CONFIRMATION_MODAL_TITLE = () => `Are you sure?`;
 export const DELETE_CONFIRMATION_MODAL_SUBTITLE = (name?: string | null) =>
   `You want to remove ${name} from this organization`;
+export const PARSING_ERROR = () =>
+  "Syntax Error: Unable to parse code, please check error logs to debug";
+export const PARSING_WARNING = () =>
+  "Linting Errors: Please remove linting errors for better experience";
```
Comment (hetunandu) [inline app/client/src/constants/messages.ts:429]:
```suggestion
  "Linting Errors: Please resolve linting errors for better experience";
```

File: app/client/src/sagas/EvaluationsSaga.ts [hunk 2]
```diff
@@ -181,15 +181,19 @@ export function* clearEvalPropertyCache(propertyPath: string) {
...
     worker.request,
     EVAL_WORKER_ACTIONS.PARSE_JS_FUNCTION_BODY,
     {
       body,
       jsAction,
     },
   );
-  return parsedObject;
+  const { errors, evalTree, result } = workerResponse;
+  yield put(setEvaluatedTree(evalTree, []));
```
Comment (hetunandu) [inline app/client/src/sagas/EvaluationsSaga.ts:194]: Sending an empty array of differences will update the whole tree. Consider updating just the __evaluation__ object

File: app/client/src/sagas/JSPaneSagas.ts [hunk 3]
```diff
@@ -270,28 +263,36 @@ function* handleExecuteJSFunctionSaga(
   const { action } = data.payload;
```
Comment (hetunandu) [inline app/client/src/sagas/JSPaneSagas.ts:263]:
```suggestion
  const { action, collectionName } = data.payload;
```

File: app/client/src/sagas/JSPaneSagas.ts [hunk 4]
```diff
@@ -270,28 +263,36 @@ function* handleExecuteJSFunctionSaga(
...
-  if (triggers && triggers.length) {
-    yield all(
-      triggers.map((trigger: ActionDescription) =>
-        call(executeActionTriggers, trigger, EventType.ON_CLICK),
-      ),
+  try {
+    const { result, triggers } = yield call(
+      executeFunction,
+      data.payload.collectionName,
+      data.payload.action,
```
Comment (hetunandu) [inline app/client/src/sagas/JSPaneSagas.ts:270]:
```suggestion
      collectionName,
      action,
```

File: app/client/src/sagas/PostEvaluationSagas.ts [hunk 5]
```diff
@@ -90,11 +90,15 @@ function logLatestEvalPropertyErrors(
...
-        ) {
-          evalWarnings.push(err);
+        if (err.severity === Severity.WARNING) {
+          if (
+            !isJSAction(entity) &&
+            !errorCodesToIgnoreInDebugger.includes(err.code || "")
+          ) {
+            evalWarnings.push(err);
+          } else {
+            evalWarnings.push(err);
```
Comment (hetunandu) [inline app/client/src/sagas/PostEvaluationSagas.ts:100]: Same effect in both if and else

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/client/src/sagas/PostEvaluationSagas.ts (hunk 5 above)
Comment (ApekshaBhosale) [reply app/client/src/sagas/PostEvaluationSagas.ts:100]: we need to filter out  errorCodesToIgnoreInDebugger = ["W117"]; types of errors for entity other than jsAction

**Review by hetunandu — APPROVED** [review]

**Pull request merged**